airflow/www/views.py

```diff
@@ -1773,8 +1856,11 @@ def recurse_nodes(task, visited):
             external_log_name=external_log_name)
 
     @expose('/graph')
-    @has_dag_access(can_dag_read=True)
-    @has_access
+    @auth.has_access([
+        (permissions.ACTION_CAN_READ, permissions.RESOURCE_DAG),
+        (permissions.ACTION_CAN_READ, permissions.RESOURCE_TASK_INSTANCE),
+        (permissions.ACTION_CAN_READ, permissions.RESOURCE_TASK_LOG),
```

Comment:
We should create a follow up PR to not make read task logs a requirement -- and if the permission is missing just don't do the `if task_log_reader.supports_external_link` bit.